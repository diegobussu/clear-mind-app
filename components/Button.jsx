import React from 'react';
import { TouchableOpacity, Text } from 'react-native';

const Button = ({ text, onPress }) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      className="bg-primary-purple mt-8 py-2 rounded-full shadow flex items-center justify-center w-48 mx-auto"
    >
      <Text className="text-white font-sf-bold text-lg">{text}</Text>
    </TouchableOpacity>
  );
};

export default Button;
