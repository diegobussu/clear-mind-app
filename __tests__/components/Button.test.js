import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

describe('Button Component', () => {
  it('rendu du texte', () => {
    const component = renderer.create(
      <Button text="Test Button" onPress={() => {}} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('onPress fonctionnel', () => {
    const onPressMock = jest.fn();
    const component = renderer.create(
      <Button text="Test Button" onPress={onPressMock} />
    );
    const buttonInstance = component.root.findByType(Button);

    buttonInstance.props.onPress();

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });
});
