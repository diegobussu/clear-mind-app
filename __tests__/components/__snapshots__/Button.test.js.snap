// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component rendu du texte 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6331FF",
      "borderBottomLeftRadius": 9999,
      "borderBottomRightRadius": 9999,
      "borderTopLeftRadius": 9999,
      "borderTopRightRadius": 9999,
      "display": "flex",
      "justifyContent": "center",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 32,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingTop": 8,
      "shadowColor": "rgba(0, 0, 0, 0.1)",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 6,
      "width": 192,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
        },
        {
          "fontFamily": "Qs-SemiBold",
        },
        {
          "fontSize": 18,
          "lineHeight": 28,
        },
      ]
    }
  >
    Test Button
  </Text>
</View>
`;
